dali/operators/reader/loader/webdataset_loader.cc

```diff
@@ -0,0 +1,417 @@
+    out.config_metadata.push_back(ComponentConfig(std::move(extension), size));
+    size = 0;
+  }
+  DALI_ENFORCE(out.config_metadata.size() > 0_uz,
+               make_string("Malformed index file at ", config_path,
+                             " - no extensions provided for the sample"));
+  return out;
+}
+
+inline std::vector<SampleConfig> ParseConfig(MissingExtBehavior missing_component_behavior,
+                                             const std::string& config_path,
+                                             std::vector<std::set<std::string>>& ext) {
+  std::ifstream config(config_path);
+  int64_t config_size;
+  size_t sample_config_num = 0;  // for config validity check
+  config >> config_size >> sample_config_num;
+  DALI_ENFORCE(sample_config_num > 0_uz,
+               "Empty index file at " + config_path);  // config validity check
+
+  std::vector<SampleConfig> out;
+  out.reserve(sample_config_num);
+
+  for (size_t sample_index = 0; sample_index < sample_config_num; sample_index++) {
+    SampleConfig new_sample = ParseSampleConfig(config, config_path);
+    if (!out.empty()) {
+      out.back().end_offset = std::min(out.back().end_offset, new_sample.start_offset);
+    }
+
+    // filtering out samples without the required extensions
+    bool discard = false;
+    if (!std::all_of(ext.begin(), ext.end(), [&](std::set<std::string> extension_set) {
+          for (auto& component_config : new_sample.config_metadata) {
+            if (extension_set.count(component_config.ext)) {
+              return true;
+            }
+          }
+          return false;
+        })) {
+      switch (missing_component_behavior) {
+        case MissingExtBehavior::Skip:
+          discard = true;
+          break;
+        case MissingExtBehavior::Raise:
+          DALI_FAIL("Underful sample detected in index file at " + config_path);
+        default:
+          break;
+      }
+    }
+    if (discard) {
+      continue;
+    }
+
+    out.push_back(std::move(new_sample));
+  }
+  if (out.size()) {
+    out.back().end_offset = std::min(out.back().end_offset, config_size);
+  }
+  return out;
+}
+
+}  // namespace wds
+}  // namespace detail
+
+inline std::string SupportedTypesListGen() {
+  std::stringstream out;
+  for (auto& dtype : detail::wds::kSupportedTypes) {
+    out << dtype << ',';
+  }
+  if (!detail::wds::kSupportedTypes.empty()) {
+    out.unget();
+  }
+  return out.str();
+}
+
+WebdatasetLoader::WebdatasetLoader(const OpSpec& spec)
+    : Loader(spec),
+      uris_(spec.GetRepeatedArgument<std::string>("uris")),
+      configs_(spec.GetRepeatedArgument<std::string>("configs")),
+      missing_component_behavior_(detail::wds::ParseMissingExtBehavior(
+          spec.GetArgument<std::string>("missing_component_behavior"))) {
+  DALI_ENFORCE(uris_.size() == configs_.size(),
+               "Number of uris does not match the number of config files");
```

Comment:
I wonder if we should add a clear indicato that this error comes from webdataset reader. But this is probably a question to all errors.

Reply:
I think it does somewhat since it prints the stacktrace?

Reply:
Long term we probably want to turn that off. But this is outside of this PR scope. Just food for thoughts.